tf/decode_training.py

```diff
@@ -2095,11 +2103,18 @@ def main(args):
             chunkdata = f.read()
             if chunkdata[0:4] == b'\1\0\0\0':
                 print("Invalid version")
-            elif chunkdata[0:4] == VERSION3:
-                #print("debug Version3")
-                for i in range(0, len(chunkdata), V3_BYTES):
-                    ts = TrainingStep(3)
-                    ts.display_v2_or_v3(i//V3_BYTES, chunkdata[i:i+V3_BYTES])
+            elif chunkdata[0:4] in {VERSION4, VERSION3}:
+                version = chunkdata[0:4]
+                if version == VERSION3:
+                    record_size = V3_BYTES
+                else:
+                    record_size = V4_BYTES
+                for i in range(0, len(chunkdata), record_size):
+                    ts = TrainingStep(4 if version == VERSION4 else 3)
+                    record = chunkdata[i:i+record_size]
+                    if chunkdata[0:4] == VERSION3:
+                        record += 16 * b'\x00'
+                    ts.display_v2_or_v3(i//record_size, record)
```

Comment:
function name seems stale.

Reply:
v2 support doesn't seem to exist in chunkparser anymore, so removing that branch entirely.